src/OpenTracing/Exceptions/SpanAlreadyFinished.php

```diff
@@ -16,7 +16,11 @@
  */
 final class SpanAlreadyFinished extends DomainException
 {
-    public static function create(Span $span)
+    /**
+     * @param Span $span
+     * @return SpanAlreadyFinished
+     */
+    public static function forSpan(Span $span)
```

Comment:
hm, the other cases are also in the similar category. I'd say it's an open question to the spec, which today says calling write methods on the span after it's finished leads to "undefined behavior", but it would be better if it said "undefined behavior but no exceptions".

it's an odd situation - we don't want people to misuse the API and write to spans after finishing them, but we also don't want the applications to crash by throwing exceptions from otherwise safe methods. 

For example, in Jaeger tracers calling span.SetTag after finishing actually *will* store the tag, but if the span has been already reported to the backend the tag will be lost. And because we synchronize when reading/writing span data, calling all write methods after Finish is actually safe.